daemon/cmd/devices.go

```diff
@@ -100,36 +98,40 @@ func (dm *DeviceManager) Detect() error {
 		detectIPv6MCastDev = false
 	}
 
-	k8sNodeDev := ""
-	k8sNodeLink, err := findK8SNodeIPLink()
-	if err == nil {
-		k8sNodeDev = k8sNodeLink.Attrs().Name
-		dm.devices[k8sNodeDev] = struct{}{}
-	}
+	if detectDirectRoutingDev || detectIPv6MCastDev {
+		k8sNodeDev := ""
+		k8sNodeLink, err := findK8SNodeIPLink()
+		if err == nil {
+			k8sNodeDev = k8sNodeLink.Attrs().Name
+			dm.devices[k8sNodeDev] = struct{}{}
+		} else if k8s.IsEnabled() {
```

Comment:
It might make sense to move the `k8s.IsEnabled()` check above the `detectDirectRoutingDev || detectIPv6MCastDev` check.

Reply:
We only require k8s when direct routing device or ipv6 mcast device detection is necessary, so it doesn't make sense to move it there as then device detection won't work without k8s. Though do we ever need device detection without k8s? I'm not sure. If not, then we can just make this fail early when k8s is not enabled.

Reply:
> Though do we ever need device detection without k8s?

Yes. In the `lb-only` mode when user specifies multiple devices, but doesn't specify the direct routing device. We should definitely keep the case `|devices| == 1 => dr_device = devices[0]` regardless of k8s presence.